### PR TITLE
Test-DbaSqlBuild, fix when no CU released after SP

### PR DESCRIPTION
**Pull request by niphlod**
followup of https://github.com/sqlcollaborative/dbatools/pull/2988

### CODE REVIEW

File: functions/Test-DbaSqlBuild.ps1 [hunk 1]
```diff
@@ -254,19 +254,21 @@ function Test-DbaSqlBuild {
                             $targetSP = 0
                         }
                         $targetSPName = $AllSPs[$targetSP]
-                        Write-Message -Level verbose -Message "Target SP is $targetSPName - $targetSP on $($AllSPs.Length)"
+                        Write-Message -Level Debug -Message "Target SP is $targetSPName - $targetSP on $($AllSPs.Length)"
```
Comment (ClaudioESSilva) [inline functions/Test-DbaSqlBuild.ps1]: Needs to be `$($AllSPs.Length -1)`

File: functions/Test-DbaSqlBuild.ps1 [hunk 2]
```diff
@@ -254,19 +254,21 @@ function Test-DbaSqlBuild {
...
-                        if ($targetCU -lt 0) {
-                            $targetCU = 0
+                        if ($AllCUs.Length -gt 0) {
+                            #CU after the targeted build available
+                            $targetCU = $AllCUs.Length - $ParsedMaxBehind['CU'] - 1
+                            if ($targetCU -lt 0) {
+                                $targetCU = 0
+                            }
+                            $targetCUName = $AllCUs[$targetCU]
+                            Write-Message -Level Debug -Message "Target CU is $targetCUName - $targetCU on $($AllCUs.Length)"
```
Comment (ClaudioESSilva) [inline functions/Test-DbaSqlBuild.ps1]: Needs to be `$($AllCUs.Length -1)`

File: functions/Test-DbaSqlBuild.ps1 [hunk 3]
```diff
@@ -254,19 +254,21 @@ function Test-DbaSqlBuild {
...
-                        $targetCU = $AllCUs.Length - $ParsedMaxBehind['CU'] - 1
-                        if ($targetCU -lt 0) {
-                            $targetCU = 0
+                        if ($AllCUs.Length -gt 0) {
+                            #CU after the targeted build available
+                            $targetCU = $AllCUs.Length - $ParsedMaxBehind['CU'] - 1
+                            if ($targetCU -lt 0) {
+                                $targetCU = 0
+                            }
+                            $targetCUName = $AllCUs[$targetCU]
```
Comment (ClaudioESSilva) [inline functions/Test-DbaSqlBuild.ps1]: `$targetCUName` must be clear before each execution.

File: functions/Test-DbaSqlBuild.ps1 (hunk 1 above)
Comment (niphlod) [reply functions/Test-DbaSqlBuild.ps1]: hum.... it's a debug message in "computer language", using 0-indexed numbers... in the end we want $AllSPs[$targetSP], so I don't see the point in changing the message

File: functions/Test-DbaSqlBuild.ps1 (hunk 2 above)
Comment (niphlod) [reply functions/Test-DbaSqlBuild.ps1]: same as previous

File: functions/Test-DbaSqlBuild.ps1 (hunk 3 above)
Comment (niphlod) [reply functions/Test-DbaSqlBuild.ps1]: right, better safeguarding also targetspname

File: functions/Test-DbaSqlBuild.ps1 (hunk 1 above)
Comment (ClaudioESSilva) [reply functions/Test-DbaSqlBuild.ps1]: The "problem" here is that RTM is being seen as "SP". That's why the output is 5.
For me the message `Target SP is SP4 - 4 on 5` indicates that I have an SP to apply and it's not true.
If we can have the message written in a way that doesn't offer doubts why not? 😄

File: functions/Test-DbaSqlBuild.ps1 (hunk 2 above)
Comment (ClaudioESSilva) [reply functions/Test-DbaSqlBuild.ps1]: Forget this comment. Here this is right!

File: functions/Test-DbaSqlBuild.ps1 (hunk 1 above)
Comment (niphlod) [reply functions/Test-DbaSqlBuild.ps1]: you need to look at the result, not at debug messages! `Target SP is SP4 - 4th element on 5 long array` would please you ?

Comment (ClaudioESSilva) [reply functions/Test-DbaSqlBuild.ps1]: Not really. For that just leave it as it is right now.

Comment (niphlod) [reply functions/Test-DbaSqlBuild.ps1]: we can also do `Target SP is SP4` :-P

**Comment by potatoqualitee** [comment]
YESSSSS !

**Pull request merged**